src/utility/positionCompute.ts

```diff
@@ -0,0 +1,197 @@
+/** Both tooltips and dropdown use the DOM to make sure
+ * they are fitting in a container
+ */
+
+const containsNumber = (value: string) => /\d/.test(value);
+const containsChar = (value: string, char: string) =>
+  value.indexOf(char) !== -1;
+
+export const checkIsStyleComputed = (style: CSSStyleDeclaration) => {
+  const { height, width } = style;
+  if (!height || !width) {
+    return false;
+  }
+
+  return (
+    containsNumber(height) ||
+    containsNumber(width) ||
+    !containsChar(height, '%') ||
+    !containsChar(width, '%')
+  );
+};
+
+const returnNumber = (value: number) => (isNaN(value) ? 0 : value);
+
+export const computeElementHeight = (
+  elementStyle: CSSStyleDeclaration,
+  alternativeHeight: string | null
+) => {
+  const height =
+    parseInt(elementStyle.paddingTop || '0', 10) +
+    parseInt(elementStyle.paddingBottom || '0', 10) +
+    parseInt(elementStyle.marginTop || '0', 10) +
+    parseInt(elementStyle.marginBottom || '0', 10) +
+    parseInt(alternativeHeight || elementStyle.height || '0', 10);
+  return returnNumber(height);
+};
+
+export const computeElementWidth = (
+  tooltipStyle: CSSStyleDeclaration,
+  alternativeWidth: string | null
+) => {
+  const width =
+    parseInt(tooltipStyle.paddingLeft || '0', 10) +
+    parseInt(tooltipStyle.paddingRight || '0', 10) +
+    parseInt(tooltipStyle.marginLeft || '0', 10) +
+    parseInt(tooltipStyle.marginRight || '0', 10) +
+    parseInt(alternativeWidth || tooltipStyle.width || '0', 10);
+
+  return returnNumber(width);
+};
+
+export const getAlternativeStyle = (
+  initialStyle: CSSStyleDeclaration,
+  element: EventTarget | null
+) => {
+  let innerDiv = document.createElement('div');
+  (Object.values(initialStyle) as (keyof CSSStyleDeclaration)[]).forEach(
+    property => {
+      if (
+        property &&
+        property !== 'length' &&
+        property !== 'parentRule' &&
+        !containsNumber(property.toString())
+      ) {
+        innerDiv.style[property] = initialStyle[property];
+      }
+    }
+  );
+  (element as Element).appendChild(innerDiv);
+  let { height, width } = innerDiv.getBoundingClientRect();
+  innerDiv.remove();
+
+  return { height: height.toString(), width: width.toString() };
+};
+
+export const getBoundingRect = (element: Element | null): ClientRect => {
+  if (element) {
+    return element.getBoundingClientRect();
+  } else {
+    return {
+      top: 0,
+      left: 0,
+      right: 0,
+      bottom: 0,
+      width: 0,
+      height: 0
+    };
+  }
+};
+
+export const getContainerBoundaries = (
+  container: Element | Window
+): ContainerDimensions => {
+  if (container === window) {
+    return {
+      maxHeight: window.innerHeight,
+      minHeight: 0,
+      maxWidth: window.innerWidth,
+      minWidth: 0
+    };
+  }
+  const containerRect = getBoundingRect(container as Element);
```

Comment:
I don't like these type assertions :) I believe here we know that the container is an Element and not the window !
If that is the case, we can do a type guard 

```suggestion
  const isWindow = (element: Element | Window): element is Window => container === window;

  if (container === window) {
    return {
      maxHeight: window.innerHeight,
      minHeight: 0,
      maxWidth: window.innerWidth,
      minWidth: 0
    };
  }
  const containerRect = getBoundingRect(container as Element);
```